### DISCUSSION TITLE
[Question] Difference between BlobCache.ForcedDateTimeKind and BlobCache.UserAccount.ForcedDateTimeKind

### DISCUSSION THREAD

**Discussion by Tviljan** [reactiveui/Akavache#769] [Q&A]
What is difference between BlobCache.ForcedDateTimeKind and BlobCache.UserAccount.ForcedDateTimeKind?
If I set BlobCache.ForcedDateTimeKind it does not affect BlobCache.UserAccount.ForcedDateTimeKind

Should BlobCache.ForcedDateTimeKind force the DateTimeKind on all locations (i.e. LocalMachine, UserAccount, Secure and InMemory)

Using Akavache 7.2.1 on Xamarin.Android

**Comment by EmilAlipiev** [comment]
Good question. I have just came to create similar question and saw this. Did you make simple test? based on my test with UWP, it is the case as you described but i need to check with Android and Ios also.

**Comment by EmilAlipiev** [comment]
Actually this is really not optimal implementation I believe. why would akavache should change the datetime? it should just take convert to tick as it is. If user wants to save it as UTC, it should be the responsibility of the App, not Akavache. Now i have a conflict. there are some I want to save as UTC, some as Local but Akavache restrict me as all or nothing. 

Of course here we can always save as Local as it will not change the anything but Akavache default is UTC. That should be changed to local not UTC. because for the new users, this is important thing and if someone doesnt pay attention will get wrong implementation and once the data is saved, it is hard to change.

**Reply by glennawatson** [reply]
It's due to the fact we store using bson not json and the reader writer has some quirks

https://github.com/reactiveui/Akavache/blob/1dd42aa4ff08a4366e0d16c7b2b219e6739e7edb/src/Akavache.Core/BlobCache/BlobCache.cs#L153

**Comment by EmilAlipiev** [comment]
I understand this, went through the code already but does it not make sense that Local should be default. Because I handle on the app if my datetime is Local or UTC should be and Package shouldnt handle UTC for me. So basically if my Local datetime is 
09.03.2022 14:01 and UTC is 09.03.2022 13:01.. and I want to save it as UTC I convert my Datetime to 09.03.2022 13.01 in the app but then Akavache save this as 09.03.2022 12:01 as ticks. That will be bit surprising for people dont see this warning.

**Reply by glennawatson** [reply]
The default won't be changing but maybe rewording the docs can help https://github.com/reactiveui/Akavache#datetimedatetimeoffset-considerations
